benchmark/testsystems/bookkeepers.py

```diff
@@ -220,30 +196,49 @@ def sample_v_given_x(self, x):
         self.simulation.context.applyVelocityConstraints(self.constraint_tolerance)
         return get_velocities(self.simulation)
 
-    def accumulate_shadow_work(self, x_0, v_0, n_steps, store_potential_energy=False):
+    def accumulate_shadow_work(self, x_0, v_0, n_steps, box_vectors=None, store_potential_energy=False, store_W_shad_trace=False):
         """Run the integrator for n_steps and return the change in energy - the heat."""
-        get_energy = lambda: get_total_energy(self.simulation)
+        get_energy = lambda: get_total_energy(self.simulation).value_in_unit(W_unit)
         get_potential = lambda: get_potential_energy(self.simulation).value_in_unit(W_unit)
         get_heat = lambda: self.simulation.integrator.getGlobalVariableByName("heat")
 
-        set_positions(self.simulation, x_0)
-        set_velocities(self.simulation, v_0)
-
         result = {}
 
+        try:
+            set_positions(self.simulation, x_0, box_vectors=box_vectors)
```

Comment:
I'm confused by the inconsistent use of `context.setPositions` and `set_positions`. Can we stick to one or the other?